httpcore/_async/connection.py

```diff
@@ -162,3 +164,76 @@ async def aclose(self) -> None:
         async with self.request_lock:
             if self.connection is not None:
                 await self.connection.aclose()
+
+
+class AsyncSOCKSConnection(AsyncHTTPConnection):
```

Comment:
A bit wary of the subclassing dependency here...

I understand the intent that "a SOCKS connection is just an HTTP connection, but connected differently". But since "open a socket" seems to be the *only* step that changes, and considering we have a `socket` parameter on `AsyncHTTPConnection`, would it make sense for the "open a SOCKS proxy socket" operation to be performed by an `AsyncSOCKSProxy` class?

In other words, instead of a separate connection, we'd have a new proxy connection pool class that opens a socket explicitly before instantiating and saving a connection.

Dunno how feasible that is and I haven't went through actually implementing this, and perhaps you've already thought about that, but just throwing ideas out there! :)

Reply:
You mean to replace inheritance by aggregation?

Reply:
Sure, never mind, I see that #51 took this approach too. :-)